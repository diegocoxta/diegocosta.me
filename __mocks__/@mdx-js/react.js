module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
