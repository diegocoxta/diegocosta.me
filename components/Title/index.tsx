import styles from './styles.module.css';

export default function Title({ children }: React.PropsWithChildren): React.ReactElement {
  return (
    <h2 className={styles.container} data-testid="title">
      {children}
    </h2>
  );
}
