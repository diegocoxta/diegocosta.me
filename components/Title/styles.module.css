.container {
  color: var(--main-title-color);
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;

  a {
    color: var(--main-title-color);
    text-decoration: none;

    &:hover,
    &:focus {
      border-bottom: 1px solid var(--main-title-color);
      outline: none;
    }
  }
}
