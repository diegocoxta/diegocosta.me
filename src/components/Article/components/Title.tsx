import React from 'react';
import styled from 'styled-components';

import { Link } from '~/utils/i18n';

const Container = styled(Link).attrs((props) => ({
  as: props.to ? Link : 'h2',
}))`
  color: ${({ theme }) => theme.titleColor};
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: bold;

  :hover,
  :focus {
    border-bottom: ${({ theme, to }) => to && `1px solid ${theme.titleColor}`};
    outline: none;
  }
`;

export interface TitleProps {
  title: string;
  url?: string | null;
}

export default function Title(props: TitleProps) {
  return (
    <Container to={props.url} data-testid="article-header-title">
      {props.title}
    </Container>
  );
}
