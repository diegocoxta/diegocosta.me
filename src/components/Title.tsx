import styled from 'styled-components';
import { Link } from 'gatsby';

export default styled(Link).attrs((props) => ({
  as: props.to ? Link : 'h2',
  'data-testid': 'title',
}))`
  color: ${({ theme }) => theme.titleColor};
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;

  &:hover,
  &:focus {
    border-bottom: ${({ theme, to }) => to && `1px solid ${theme.titleColor}`};
    outline: none;
  }
`;
