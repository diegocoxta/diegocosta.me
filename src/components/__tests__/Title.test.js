import React from 'react';
import { render } from '@testing-library/react';

import Title from '../Title';

describe('<Title />', () => {
  it('renders properly', () => {
    const { baseElement, getByText } = render(<Title>My page title</Title>);

    expect(getByText('My page title')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it('should render as link', () => {
    const { baseElement, getByText, getByTestId } = render(<Title to="https://google.com">My page title</Title>);

    expect(getByText('My page title')).toBeTruthy();
    expect(getByTestId('title').href).toBe('https://google.com/');
    expect(baseElement).toMatchSnapshot();
  });
});
