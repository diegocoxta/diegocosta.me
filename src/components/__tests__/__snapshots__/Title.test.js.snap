// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders properly 1`] = `
.c0 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c0:hover,
.c0:focus {
  outline: none;
}

<body>
  <div>
    <h2
      class="c0"
      data-testid="title"
    >
      My page title
    </h2>
  </div>
</body>
`;

exports[`<Title /> should render as link 1`] = `
.c0 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c0:hover,
.c0:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

<body>
  <div>
    <a
      class="c0"
      data-testid="title"
      href="https://google.com"
    >
      My page title
    </a>
  </div>
</body>
`;
